Add label component (#220)
primer/brand: apps/docs/content/components/Label.mdx

---
title: Label
status: Experimental

> figma: 'https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=6084-28401'

Reviewer: This goes to Button page in Figma. Can we repoint to Label?

Author: Do you have the Figma branch updated? I just double-checked that and it goes to the Label page.

Reviewer: Huh, yeah it goes there when i open in browser, but in app it loaded button. 🤷 .. guess it's fine and maybe my app is buggy.

source: https://github.com/primer/brand/blob/main/packages/react/src/Label/Label.tsx
storybook: '/brand/storybook/?path=/story/components-label--default'

> description: Use a Label to indicate the status of an object or to categorize it with metadata that describes it.

Reviewer: Could we use a similar description to PCSS? 

```suggestion
Use Labels to add metadata or indicate status of items and navigational elements.
```

Author: I think the usage in Brand is a bit different as we don't put labels for navigational items, but headings/titles mainly.

What about something in between and simpler?

```suggestion
description: Use Labels to add metadata or indicate an object's status.
```

```jsx live
<Stack direction="horizontal" alignItems="flex-start" padding="none">
  <Label size="small">Small</Label>
  <Label size="medium">Medium</Label>
  <Label size="large">Large</Label>
</Stack>
```

### Color

Use the `color` prop to change the color of the `Label` component. The `default` color is recommended for use with neutral backgrounds.

```jsx live
<Stack direction="horizontal" alignItems="flex-start" padding="none">
  <Label color="blue">Blue</Label>
  <Label color="red">Red</Label>
  <Label color="green-blue">Green blue</Label>
  <Label color="red">Red</Label>
  <Label color="pink-blue">Pink blue</Label>
  <Label color="orange">Orange</Label>
</Stack>
```

### Leading visual

You can add a leading icon to enhance the visual context. It is recommended you use an [Octicon](https://primer.style/octicons) here.

```jsx live
<Stack direction="horizontal" alignItems="flex-start" padding="none">
  <Label color="yellow" leadingVisual={StarIcon}>
    Starred
  </Label>
  <Label color="blue-purple" leadingVisual={CopilotIcon}>
    Copilot
  </Label>
</Stack>
```

## Component props

> ### Button

Reviewer: ```suggestion
### Label
```

Author: oopsie!
